packages/core/state/interaction/logics.ts

```diff
@@ -625,21 +626,44 @@ const copyFilesLogic = createLogic({
                     status === "DOWNLOAD_IN_PROGRESS" ||
                     status === "DOWNLOAD_STARTED"
                 ) {
-                    successfulFiles.push(fileId);
+                    const file = fileDetails.find((f) => f.id === fileId);
+                    const isAlreadyCached = file?.annotations.some(
+                        ({ name, values }) =>
+                            name === "Should Be in Local Cache" && values[0] === true
```

Comment:
we might want to use the `AnnotationName.SHOULD_BE_IN_LOCAL` constant here and elsewhere (saw that the full string is also used in other places) if possible